modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/actionItems/formatActionURL.ts

```diff
@@ -37,12 +44,52 @@ const formatActionURL = function (url: string | undefined, item: any): string {
 		)
 	);
 
-	return replacedURL.replace(new RegExp('(%7B.*?%7D)', 'mg'), (matched) =>
+	replacedURL.replace(new RegExp('(%7B.*?%7D)', 'mg'), (matched) =>
 		getValueFromItem(
 			item,
 			matched.substring(3, matched.length - 3).split('.')
 		)
 	);
+
+	if (target === 'link' && replacedURL.includes('?')) {
+		const redirectionURL = window.location.href;
+		const searchParams = new URLSearchParams(
+			replacedURL.slice(replacedURL.indexOf('?'))
+		);
+
+		const p_p_id = searchParams.get('p_p_id');
+		const backURLParam = `_${p_p_id}_backURL`;
+		const redirectParam = `_${p_p_id}_redirect`;
```

Comment:
This looks like it will work most of the time, and will work for Users MVP. But I believe there are some edge cases we can cover with a slightly different algorithm:
1. Parse all params for `backURL`. 
2. If there is a `backURL`, set value of that param, keeping the param name _exactly as it is_.
3. If there is no `backURL`, only then look for `p_p_id`, and construct param as `_${p_p_id}_backURL`.
4. If there is no `p_p_id` and no `backURL`, do nothing.

This would work in the most common use case, but also in some edge cases:
1. `p_p_id` is not set. This may be if there is no portlet on target page, but a fragment or some custom injected solution.
2. `backURL` and `p_p_id` are different. This may be if there are multiple portlets on the page, or if one portlet is included in another, and inner portlet contains back link.

Reply:
@markocikos :thinking: I would say that this is the way it works now. 

[First condition](./portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/dataset) looks for `_backURL` || `_redirect`.
[Otherwise will check `p_p_id` ](https://github.com/liferay-frontend/liferay-portal/pull/4154/files#diff-1931ce4d696e18393ca3752a004570d2a2ef5563ba7b008b2e1eb37ea611912bR77) and set both parameters.

If the URL doesn't meet any condition, it will do nothing. We have a [test](https://github.com/liferay-frontend/liferay-portal/pull/4154/files#diff-1931ce4d696e18393ca3752a004570d2a2ef5563ba7b008b2e1eb37ea611912bR77) that covers this case.

Reply:
Your link the first condition is broken. But looking at code, you indeed have `p_p_id` handling correct. But my edge cases are for `backURL` without a prefix underscore, with no namespace, or different namespace:
```js
[base url]?p_p_id=[portlet ns]&[portlet ns]_backURL= // Covered
[base url]?p_p_id= // Covered
[base url]?[portlet ns]_backURL= // Not covered, but should not happen
[base url]?backURL= // Not covered. This could be on a page with just a fragment.
[base url]?p_p_id=[portlet ns]&backURL= // Not covered, might cause a bug. This could be when fragment and portlet are on the same page.
[base url]?p_p_id=[portlet ns]&[other portlet ns]_backURL= // Not covered. Might happen when a portlet is injected in another portlet.
```